helpers/url.go

```diff
@@ -147,47 +147,98 @@ func MakePermalink(host, plink string) *url.URL {
 }
 
 // AbsURL creates a absolute URL from the relative path given and the BaseURL set in config.
-func AbsURL(path string) string {
-	url, err := url.Parse(path)
+func AbsURL(in string, addLanguage bool) string {
+	url, err := url.Parse(in)
 	if err != nil {
-		return path
+		return in
 	}
 
-	if url.IsAbs() || strings.HasPrefix(path, "//") {
-		return path
+	if url.IsAbs() || strings.HasPrefix(in, "//") {
+		return in
 	}
 
 	baseURL := viper.GetString("BaseURL")
-	if strings.HasPrefix(path, "/") {
+	if strings.HasPrefix(in, "/") {
 		p, err := url.Parse(baseURL)
 		if err != nil {
 			panic(err)
 		}
 		p.Path = ""
 		baseURL = p.String()
 	}
-	return MakePermalink(baseURL, path).String()
+
+	if addLanguage {
+		prefix := getLanguagePrefix()
+
+		if prefix != "" {
+			addSlash := in == "" || strings.HasSuffix(in, "/")
+			in = path.Join(prefix, in)
+
+			if addSlash {
+				in += "/"
+			}
+		}
+	}
+	return MakePermalink(baseURL, in).String()
+}
+
+func getLanguagePrefix() string {
+	defaultLang := viper.GetString("DefaultContentLanguage")
+	defaultInSubDir := viper.GetBool("DefaultContentLanguageInSubdir")
+
+	if !viper.GetBool("Multilingual") {
+		return ""
+	}
+	currentLang := viper.Get("CurrentContentLanguage").(*Language).Lang
```

Comment:
I need to panic if it isn't there (which suggests a programming error anyhow), so it might as well be a panic with a nil pointer.